src/node_crypto.cc

```diff
@@ -4934,46 +5637,16 @@ class ECKeyPairGenerationConfig : public KeyPairGenerationConfig {
   const int param_encoding_;
 };
 
-enum PKEncodingType {
-  // RSAPublicKey / RSAPrivateKey according to PKCS#1.
-  PK_ENCODING_PKCS1,
-  // PrivateKeyInfo or EncryptedPrivateKeyInfo according to PKCS#8.
-  PK_ENCODING_PKCS8,
-  // SubjectPublicKeyInfo according to X.509.
-  PK_ENCODING_SPKI,
-  // ECPrivateKey according to SEC1.
-  PK_ENCODING_SEC1
-};
-
-enum PKFormatType {
-  PK_FORMAT_DER,
-  PK_FORMAT_PEM
-};
-
-struct KeyPairEncodingConfig {
-  PKEncodingType type_;
-  PKFormatType format_;
-};
-
-typedef KeyPairEncodingConfig PublicKeyEncodingConfig;
-
-struct PrivateKeyEncodingConfig : public KeyPairEncodingConfig {
-  const EVP_CIPHER* cipher_;
-  // This char* will be passed to OPENSSL_clear_free.
-  std::shared_ptr<char> passphrase_;
-  unsigned int passphrase_length_;
-};
-
 class GenerateKeyPairJob : public CryptoJob {
  public:
   GenerateKeyPairJob(Environment* env,
                      std::unique_ptr<KeyPairGenerationConfig> config,
                      PublicKeyEncodingConfig public_key_encoding,
-                     PrivateKeyEncodingConfig private_key_encoding)
+                     PrivateKeyEncodingConfig&& private_key_encoding)
     : CryptoJob(env),
     config_(std::move(config)),
     public_key_encoding_(public_key_encoding),
-    private_key_encoding_(private_key_encoding),
+    private_key_encoding_(std::move(private_key_encoding)),
```

Comment:
IIUC you need to [`std::forward`](https://en.cppreference.com/w/cpp/utility/forward) lvalue ref args